typescript/test/util.ts

```diff
@@ -0,0 +1,103 @@
+import * as assert from 'assert';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+
+const BASELINE_EXTENSION = '.baseline';
+export function compareToBaseline(outpurDir: string, baselineDir: string) {
+  // put all baseline files into fileStack
+  let fileStack: string[] = [];
+  const dirStack: string[] = [];
+  putAllBaselineFiles(baselineDir, fileStack, dirStack);
+  // store every item (file or directory with full path in output dir and baseline dir) in stack, pop once a time
+  const protoItemStack: Item[] = [];
+  putItemToStack(protoItemStack, outpurDir, baselineDir);
+  while (protoItemStack.length !== 0) {
+    const item = protoItemStack[0];
+    protoItemStack.shift();
+    // if item is a file, compare it with baseline
```

Comment:
seems array.pop() is removing element from the end, debugged a little bit to find it. So use shift here to make sure we remove the first element every time.

Reply:
Then it's a queue and not a stack

Reply:
yes, then I probably should get item from the end.

Reply:
It does not really matter if you take it from the beginning or from the end (it only affects the order of files but at the end produces the same results), but the variable name should match the way you're working with it.